### PR TITLE
Add possibility to do auth for Aqua SaaS near default auth Aqua.

### PR DESCRIPTION
**Pull request by Ukhanskyi**
Trying to connect to Aqua SaaS, near existing authentication logic.

[GitHub Issue](https://github.com/KennaSecurity/toolkit/issues/550#issue-2362548416)

### CODE REVIEW

**Comment by Ukhanskyi** [comment]
@jgarber, @shotop, @scotttesler, @JaiHirsch, Hi everyone 👋🏼)
Would it be possible for someone to review these changes for the Aqua connector and merge them if everything is correct?

**Comment by Ukhanskyi** [comment]
@jgarber-cisco, or maybe you could to check this code during your business hours.

File: tasks/connectors/aqua/lib/aqua_helper.rb [hunk 1]
```diff
@@ -5,30 +5,77 @@
...
-        print_debug "Getting Auth Token"
-        aqua_auth_api = "#{aqua_url}/api/v1/login"
-        # auth_headers = { "content-type" => "application/json",
-        #            "accept" => "application/json" }
-        # auth_body = { "id" => "administrator",
-        #              "password" => "My@rvgicmx1" }
-
-        @headers = { "Content-Type" => "application/json" }
-        payload = {
-          "id": username.to_s,
```
Comment (jgarber-cisco) [inline tasks/connectors/aqua/lib/aqua_helper.rb:18]: I don't have access to the API docs to know: is `email` on line 31 an acceptable substitute for `id` when using Aqua on-prem?

**Review by jgarber-cisco — APPROVED** [review]
I have a question about whether this breaks on-prem, but otherwise, LGTM!

File: tasks/connectors/aqua/lib/aqua_helper.rb [hunk 2]
```diff
@@ -5,30 +5,77 @@
...
-          "password": password.to_s
-        }
+        if aqua_url.include?("cloud.aquasec.com")
```
[the remark this answers is not in the document]
Comment (jgarber-cisco) [reply tasks/connectors/aqua/lib/aqua_helper.rb:12]: Oh, that's a good suggestion! +1

File: tasks/connectors/aqua/aqua.rb [hunk 3]
```diff
@@ -102,6 +103,8 @@
         token = aqua_get_token(aqua_url, username, password)
         fail_task "Unable to authenticate with Aqua, please check credentials" unless token
 
+        aqua_url = get_wp_url(token) if aqua_url.include?("cloud.aquasec.com")
```
[the remark this answers is not in the document]
Comment (jgarber-cisco) [reply tasks/connectors/aqua/aqua.rb:106]: Probably you'll want to abstract the two checks for `cloud.aquasec.com` into a method as your url check gets a little more sophisticated.

**Comment by jgarber-cisco** [comment]
@Ukhanskyi Rebase to pick up the security upgrades from `main`

**Review by jgarber-cisco — CHANGES_REQUESTED** [review]
Rebase, look at on-prem, and make cloud url checking safer.

File: tasks/connectors/aqua/lib/aqua_helper.rb (hunk 1 above)
Comment (Ukhanskyi) [reply tasks/connectors/aqua/lib/aqua_helper.rb:18]: Me too, so I've reverted id when using Aqua on-prem

**Comment by Ukhanskyi** [comment]
> Rebase, look at on-prem, and make cloud url checking safer.

Rebased on the main branch and squashed all commits. I returned the 'id' for on-prem, and also tried to make the URL check more accurate and safe. Could you review this code one more time, please

**Review by jgarber-cisco — APPROVED** [review]
Great job!

**Pull request merged**